datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/agent_signature.py

```diff
@@ -0,0 +1,31 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import click
+
+from ...utils import complete_valid_checks, get_valid_checks, has_legacy_signature
+from ..console import CONTEXT_SETTINGS, echo_failure, echo_success
+
+
+@click.command(
+    'legacy-signature',
+    context_settings=CONTEXT_SETTINGS,
+    short_help='Validate that no integration uses the legacy signature',
+)
+@click.argument('check', autocompletion=complete_valid_checks, required=False)
+def legacy_signature(check):
+    """Validate that no integration uses the legacy signature."""
+    if check:
+        checks = [check]
+    else:
+        checks = sorted(get_valid_checks())
+
+    has_failed = False
+    for check in checks:
+        if has_legacy_signature(check):
+            has_failed = True
+            echo_failure(f'Check {check} uses legacy agent signature.')
```

Comment:
Might helpful to print the line text and line number

Reply:
Good idea! I can add that

Reply:
I ended up printing both the file name and line number.